### PR TITLE
Added recipe for libsemigroups 0.2.0

### PR DESCRIPTION
**Pull request by nthiery**

### CODE REVIEW

File: recipes/libsemigroups/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+source:
+  url: {{ repo }}/releases/download/v{{ version }}/{{ name }}-{{ version }}.tar.gz
+  md5: {{ md5 }}
+  #git_rev: v{{ version }}
+  #git_url: {{ repo }}.git
+  #git_tag: autotools
+
+build:
+  number: 0
```
Comment (isuruf) [inline recipes/libsemigroups/meta.yaml]: If windows is not supported, add `  skip : True  # [win]`

File: recipes/libsemigroups/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
```
Comment (isuruf) [inline recipes/libsemigroups/meta.yaml]: Add `toolchain` as a build requirement.

File: recipes/libsemigroups/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+test:
+  commands:
+    - test -d ${PREFIX}/include/libsemigroups  # [unix]
+    - test -f ${PREFIX}/include/libsemigroups/semigroups.h  # [unix]
```
Comment (isuruf) [inline recipes/libsemigroups/meta.yaml]: Can you check for the existence of the static and shared libraries?

File: recipes/libsemigroups/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    libsemigroups is a C++ library for semigroups and monoids; it is partly based on 
+    [Algorithms for computing finite semigroups](https://www.irif.fr/~jep/PDF/Rio.pdf), 
+    [Expository Slides](https://www.irif.fr/~jep/PDF/Exposes/StAndrews.pdf), and 
+    [Semigroupe 2.01](https://www.irif.fr/~jep/Logiciels/Semigroupe2.0/semigroupe2.html) by Jean-Eric Pin.
+  doc_url: {{ home }}
+  dev_url: {{ repo }}
+
+extra:
+  recipe-maintainers:
+    - james-d-mitchell
```
Comment (isuruf) [inline recipes/libsemigroups/meta.yaml]: Did you ask `james-d-mitchell` whether he wants to be a maintainer for this conda package?

File: recipes/libsemigroups/meta.yaml (hunk 2 above)
Comment (nthiery) [inline recipes/libsemigroups/meta.yaml]: Ok, I was wondering about this. I had seen it in a couple recipes, but I found the "toolchain"  conda package README rather uninformative. Thanks for the tip!

File: recipes/libsemigroups/meta.yaml (hunk 1 above)
Comment (nthiery) [inline recipes/libsemigroups/meta.yaml]: I havent yet checked the CI report. In principle, libsemigroups is supported in Windows, via cygwin. What's the proper way to specify that?

File: recipes/libsemigroups/meta.yaml (hunk 3 above)
Comment (nthiery) [inline recipes/libsemigroups/meta.yaml]: Ok. I wasn't sure how to do this in a portable way. Do you have a recommendation or example recipe?

File: recipes/libsemigroups/meta.yaml (hunk 4 above)
Comment (nthiery) [inline recipes/libsemigroups/meta.yaml]: Yes, he is sitting next to me :-)

**Comment by nthiery** [comment]
Thanks a lot Isuru for the review; that's super helpful.

File: recipes/libsemigroups/meta.yaml (hunk 3 above)
Comment (isuruf) [inline recipes/libsemigroups/meta.yaml]: https://github.com/conda-forge/eclib-feedstock/blob/master/recipe/meta.yaml#L37-L39

File: recipes/libsemigroups/meta.yaml (hunk 2 above)
Comment (isuruf) [inline recipes/libsemigroups/meta.yaml]: `toolchain` sets the compilers and `CFLAGS` `LDFLAGS` etc. for building packages in conda-forge

File: recipes/libsemigroups/meta.yaml (hunk 1 above)
Comment (isuruf) [inline recipes/libsemigroups/meta.yaml]: I don't think there is any. I assume you want to use the python bindings, if so building with MSVC is recommended.

File: recipes/libsemigroups/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+{% set name = "libsemigroups" %}
+{% set version = "0.2.1" %}
+{% set md5 = "de073733a7d389e6ce378f4586ff1f42" %}
```
Comment (nthiery) [inline recipes/libsemigroups/meta.yaml]: Ok, fixed.

File: recipes/libsemigroups/meta.yaml (hunk 3 above)
Comment (nthiery) [inline recipes/libsemigroups/meta.yaml]: Done for shared after having found an example in alps' recipe

File: recipes/libsemigroups/meta.yaml (hunk 1 above)
Comment (nthiery) [inline recipes/libsemigroups/meta.yaml]: For now, I am skipping windows.

File: recipes/libsemigroups/meta.yaml (hunk 3 above)
Comment (isuruf) [inline recipes/libsemigroups/meta.yaml]: Add `test -f $PREFIX/lib/libsemigroups.a     # [unix]` as well

File: recipes/libsemigroups/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  build:
+    - toolchain
+
+test:
+  commands:
+    - test -d ${PREFIX}/include/libsemigroups  # [unix]
+    - test -f ${PREFIX}/include/libsemigroups/semigroups.h  # [unix]
+    - test -f $PREFIX/lib/libsemigroups.dylib  # [osx]
+    - test -f $PREFIX/lib/libsemigroups.so     # [linux]
+    - test -f $PREFIX/lib/libsemigroups.a # [unix]
```
Comment (isuruf) [inline recipes/libsemigroups/meta.yaml]: This is a weird error from linter. Can you add a space before `#`?

**Comment by nthiery** [comment]
>    This is a weird error from linter. Can you add a space before #?

Oh, thanks, I had trouble indeed interpreting the message from the linter. Done.

**Comment by nthiery** [comment]
>    I don't think there is any. I assume you want to use the python
>    bindings, if so building with MSVC is recommended.

Ok, good to know. I'll discuss the matter with James to see what are
the odds to get this running with MSVC.

**Comment by isuruf** [comment]
I had a quick look at the Makefile and it seems it should be easy to write a CMakeLists.txt for a cmake build.

**Comment by nthiery** [comment]
Ok; what gain do you have in mind when suggesting using cmake?
Are you suggesting to use it on top of the existing autotools build system, or as a replacement?

**Comment by isuruf** [comment]
As a replacement or as another build system. Main advantage is that it can generate Visual Studio project files, so that you can compile with MSVC on windows. CMake also supports linux, osx and many other systems and compilers.

**Comment by nthiery** [comment]
>    As a replacement or as another build system. Main advantage is that it
>    can generate Visual Studio project files, so that you can compile with
>    MSVC on windows. CMake also supports linux, osx and many other systems
>    and compilers.

Ok. It's really disappointing that autotools don't support MSVC. We
are a bit stuck with it, for consistency with other software in the
ecosystem. For now, we are going to drop support for windows; pending
enough interest, we will see about maintaining a double build system
autotools + cmake.

Thanks @isuruf for all the wisdom, here and elsewhere in this review!

**Comment by isuruf** [comment]
@nthiery, @james-d-mitchell  welcome to conda-forge. Here's some information that you might need. http://conda-forge.github.io/docs/recipe.html#what-happens-after-the-pr-to-staged-recipes-is-merged

**Pull request merged**